daemon/CryptoImpl/crypto.cpp

```diff
@@ -508,19 +508,19 @@ void Daemon::ApiImpl::CryptoRequestQueue::handlePendingRequest(
             qCDebug(lcSailfishCryptoDaemon) << "Handling EncryptRequest from client:" << request->remotePid << ", request number:" << request->requestId;
             QByteArray encrypted;
             QByteArray data = request->inParams.size() ? request->inParams.takeFirst().value<QByteArray>() : QByteArray();
+            QByteArray iv = request->inParams.size() ? request->inParams.takeFirst().value<QByteArray>() : QByteArray();
             Key key = request->inParams.size() ? request->inParams.takeFirst().value<Key>() : Key();
             Key::BlockMode blockMode = request->inParams.size() ? request->inParams.takeFirst().value<Key::BlockMode>() : Key::BlockModeUnknown;
             Key::EncryptionPadding padding = request->inParams.size() ? request->inParams.takeFirst().value<Key::EncryptionPadding>() : Key::EncryptionPaddingUnknown;
-            Key::Digest digest = request->inParams.size() ? request->inParams.takeFirst().value<Key::Digest>() : Key::DigestUnknown;
             QString cryptosystemProviderName = request->inParams.size() ? request->inParams.takeFirst().value<QString>() : QString();
```

Comment:
Can the `m_requestProcessor` actually do anything useful with all of these parameters evaluated to defaults?

Reply:
As mentioned in the other PR, the default parameter fallbacks are purely to prevent a crash, and should never occur in practice (if the parameter doesn't exist, it would indicate that we've marshalled the asynchronous request wrongly).  The "real" fix for this is to fix the marshalling to use per-request-type structs instead of QVariantList -- but that should be done later once we've finished the API improvements, IMO.

Reply:
Ok.